game/view/sector.lua

```diff
@@ -223,11 +224,92 @@ function SectorView:setCursorPos(i,j)
   self.cursor.j = j
 end
 
+--Function checks if a target position is inside "target cone" given desired direction {di,dj}
+local function _isInCone(origin_i, origin_j, target_i, target_j, di, dj)
+  local delta_i = target_i - origin_i
+  local delta_j = target_j - origin_j
+
+  if     di == -1 and dj == 0 then   --UP
+    return (delta_j < -2*delta_i and delta_j > 2*delta_i)
+  elseif di == 0 and dj == 1 then   --RIGHT
+    return (delta_i < 2*delta_j and delta_i > -2*delta_j)
+  elseif di == 1 and dj == 0 then   --DOWN
+    return (delta_j < 2*delta_i and delta_j > -2*delta_i)
+  elseif di == 0 and dj == -1 then     --LEFT
+    return (delta_i < -2*delta_j and delta_i > 2*delta_j)
+  elseif di == -1 and dj == 1 then   --UPRIGHT
+    return (delta_i >= -2*delta_j and 2*delta_i <= -delta_j)
+  elseif di == 1 and dj == 1 then   --DOWNRIGHT
+    return (2*delta_i >= delta_j and delta_i <= 2*delta_j)
+  elseif di ==  1 and dj == -1 then   --DOWNLEFT
+    return (2*delta_i >= -delta_j and delta_i <= -2*delta_j)
+  elseif di == -1 and dj == -1 then   --UPLEFT
+    return (delta_i >= 2*delta_j and 2*delta_i <= delta_j)
+  else
+    return error("Not valid direction for cone function")
+  end
+
+end
+
 function SectorView:moveCursor(di,dj)
   if not self.cursor then return end
 
-  self.cursor.i = self.cursor.i + di
-  self.cursor.j = self.cursor.j + dj
+  local sector = self.route.getCurrentSector()
+  local queue = Queue(128)
+  local pos
+  local sector_map = {}
+
+  if self.cursor.i + di < 1 or self.cursor.i + di > sector.h or  --Check if first position is inside sector
```

Comment:
I'm pretty sure `sector` has a method for checking this. If it doesn't it's better if it's there than here.